modules/navier_stokes/src/userobjects/PINSFVRhieChowInterpolator.C

```diff
@@ -0,0 +1,113 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVRhieChowInterpolator.h"
+#include "Reconstructions.h"
+#include "NS.h"
+#include "Assembly.h"
+
+registerMooseObject("NavierStokesApp", PINSFVRhieChowInterpolator);
+
+InputParameters
+PINSFVRhieChowInterpolator::validParams()
+{
+  auto params = INSFVRhieChowInterpolator::validParams();
+  params.addClassDescription("Performs interpolations and reconstructions of body forces and "
+                             "porosity and computes the Rhie-Chow face velocities.");
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "The porosity");
+  params.addParam<unsigned short>(
+      "smoothing_layers",
+      0,
+      "The number of interpolation-reconstruction operations to perform on the porosity");
+  params.addRelationshipManager(
+      "ElementSideNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC,
+      [](const InputParameters & obj_params, InputParameters & rm_params) {
+        // We need one additional layer for the case that we have an extrapolated boundary face
+        rm_params.set<unsigned short>("layers") =
+            obj_params.get<unsigned short>("smoothing_layers") + 1;
+        rm_params.set<bool>("use_displaced_mesh") = obj_params.get<bool>("use_displaced_mesh");
+      });
```

Comment:
in a lot of use cases (maybe all) a one-term expansion would be fine since the porosity is likely piecewise-subdomain-constant. However, for a continuously varying porosity (like in MMS tests) a one-term expansion will lose second-order accuracy

Reply:
so:
- if we imagined a porosity functor that depended on a variable (like temperature), we would be in trouble right? Since we do geometric not algebraic ghosting?

- would we loose second order accuracy if for the lowest level of smoothing (like the values that get mixed, then get mixed again, then again etc) we use a one-term expansion?

Reply:
I have an error for if the user has requested smoothing layers and the functor is a variable because it's common to ask for quite a few layers of porosity smoothing; the test I added uses 20 which is an amount that @tanoret commonly uses. I think 20 layers of algebraic ghosting would be ridiculous.

> would we loose second order accuracy if for the lowest level of smoothing (like the values that get mixed, then get mixed again, then again etc) we use a one-term expansion?

Hard to say without testing but in general I would expect that at some point a first order error anywhere will pollute an otherwise second order method

Reply:
Agreed that eps being a variable is handled well. I m asking for eps(T)

Reply:
Then yes we would be in trouble. This trouble would result in numeric vector read assertion failures from libMesh in `devel` and `dbg` mode and probably a crash in other modes

Reply:
Do you think you'll be putting T dependence into porosity soon? I don't actually a good programmatic idea for creating the requisite algebraic ghosting if epsilon is a functor material property that calls variable functors

Reply:
No I dont think so. Was just thinking.
Yeah it would hurt us a lot with AD & with ghosting. 

So I guess we'd eventually need a functor_ghosting_check on top of a functor_coverage_check.